changelogs/unreleased/gh-5272-bootstrap-strategy-auto.md

```diff
@@ -0,0 +1,8 @@
+## feature/replication
+
+* **[Breaking change]** Introduce new configuration option `bootstrap_strategy`.
```

Comment:
AFAIU breaking changes aren't allowed in minor releases anymore. We should've introduced a new compat option for this change. @Totktonada note.

Reply:
We discussed this with @Mons and he asked to not introduce a comat option for this change.

Reply:
Does it work this way? Can we ruin any rule if a product meneger asks for that? (And, out of curiosity, can a product manager create a rule that he can't break?)

I would like at least to hear a motivation behind this decision.

Reply:
> Can we ruin any rule if a product manager asks for that? 

TBH I didn't know this is a rule. I thought this was for us (or for the product manager) to decide.

Reply:
Rules regarding backward compatibility and how they spread across releases are described here: https://github.com/tarantool/tarantool/discussions/6182. At time of publishing this proposal (mid 2021) we had agreement at least within authors of the document and had no complains from managers of developments teams. It states that the rules are to be applied to 2.10.0 and beyond.

A short version of this document was created for the website: https://www.tarantool.io/en/doc/latest/release/policy/

There are also some helper materials for maintainers, which were inspired by the new release policy: https://github.com/tarantool/tarantool/wiki/Maintainer's-guide. I'll cite a decision table from it:

> | Git branch | Bugfixes? | New features? | Breaking changes? |
> | ---------- | --------- | ------------- | ----------------- |
> | `1.10`     | Yes       | No            | No                |
> | `2.10`     | Yes       | No            | No                |
> | `master`   | Yes       | Yes           | No                |

I would say that all those rules are quite natural (semver alike), but I would also say that it looks like quite official.

There are internal discussions, where we're pushed to violate them, but I hope that we'll reduce negative effects here as much as possible.

You question regarding a responsible person is the really good one. Sadly, I have no answer at the moment. It seems, we're weak here. (It is better to discuss within the team.)

----

Thanks for the detailed answer! I can't judge, but for me it actually looks as a compatible change (or, well, compatible to some pretty good extent). Shouldn't we notify a user about upgrading risks in some other way, without stating that the change is incompatible?

Reply:
Thanks for your answer!

> Thanks for the detailed answer! I can't judge, but for me it actually looks as a compatible change (or, well, compatible to some pretty good extent). Shouldn't we notify a user about upgrading risks in some other way, without stating that the change is incompatible?

As for marking this change in some other way, I'd probably stick with naming it a breaking change. The user has the necessary info right next to the [breaking_change] marker and can decided whether this will break anything for him or not.

Reply:
I don't think you need both - an option and a note about a breaking change.
If you are doing a breaking change anyway, just drop the "legacy" setting and drop the option.

I agree with Mons the impact of the breaking change is low and best to just fix it. It's not a feature, it's a bug.